compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute.reservation;
+
+// [START compute_reservation_create_shared]
+
+import com.google.cloud.compute.v1.AllocationSpecificSKUReservation;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Reservation;
+import com.google.cloud.compute.v1.ReservationsClient;
+import com.google.cloud.compute.v1.ShareSettings;
+import com.google.cloud.compute.v1.ShareSettingsProjectConfig;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class CreateSharedReservation {
+  private final ReservationsClient reservationsClient;
+
+  // Constructor to inject the ReservationsClient
+  public CreateSharedReservation(ReservationsClient reservationsClient) {
+    this.reservationsClient = reservationsClient;
+  }
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // The ID of the project where you want to reserve resources
+    // and where the instance template exists.
+    String projectId = "YOUR_PROJECT_ID";
```

Comment:
Make sure to highlight the fact, that this project MUST be allowlisted in `compute.sharedReservationsOwnerProjects` org policy. See more at: https://cloud.google.com/compute/docs/instances/reservations-shared#shared_reservation_constraint